website/client/src/components/tasks/modal-controls/checklist.vue

```diff
@@ -0,0 +1,265 @@
+<template>
+  <div class="checklist-component">
+    <label v-once class="mb-1">{{ $t('checklist') }}</label>
+    <br>
+    <draggable
+      v-model="checklist"
+      :options="{
+        handle: '.grippy',
+        filter: '.task-dropdown',
+        disabled: disabled,
+      }"
+      @update="sortedChecklist"
+    >
+      <div
+        v-for="(item, $index) in checklist"
+        :key="item.id"
+        class="inline-edit-input-group checklist-group input-group"
+      >
+        <span
+          class="grippy"
+          v-html="icons.grip"
+          v-if="!disabled"
+        >
+        </span>
+
+          <checkbox :checked.sync="item.completed"
+                    :disabled="disabled"
+                    class="input-group-prepend"
+                    :class="{'cursor-auto': disabled}"
+                    :id="`checklist-${item.id}`"/>
+
+        <input
+          v-model="item.text"
+          class="inline-edit-input checklist-item form-control"
+          type="text"
+          :disabled="disabled"
+        >
+        <span
+          class="input-group-append"
+          v-if="!disabled"
+          @click="removeChecklistItem($index)"
+        >
+          <div
+            class="svg-icon destroy-icon"
+            v-html="icons.destroy"
+          ></div>
+        </span>
+      </div>
+    </draggable>
+    <div
+      class="inline-edit-input-group checklist-group input-group new-checklist"
+      v-if="!disabled"
+    >
+      <span class="input-group-prepend new-icon"
+           v-html="icons.positive">
+
+      </span>
+
+      <input
+        v-model="newChecklistItem"
+        class="inline-edit-input checklist-item form-control"
+        type="text"
+        :placeholder="$t('newChecklistItem')"
+        @keypress.enter="setHasPossibilityOfIMEConversion(false)"
+        @keyup.enter="addChecklistItem($event, true)"
+        @blur="addChecklistItem($event, false)"
+      >
+    </div>
+  </div>
+</template>
+
+<script>
+// import clone from 'lodash/clone';
+import draggable from 'vuedraggable';
+import uuid from 'uuid';
+
+import positiveIcon from '@/assets/svg/positive.svg';
+import deleteIcon from '@/assets/svg/delete.svg';
+import chevronIcon from '@/assets/svg/chevron.svg';
+import gripIcon from '@/assets/svg/grip.svg';
+import checkbox from '@/components/ui/checkbox';
+
+export default {
+  components: {
+    draggable,
+    checkbox,
+  },
+  name: 'checklist',
+  data () {
+    return {
+      checklist: this.items,
+      hasPossibilityOfIMEConversion: true,
+      newChecklistItem: null,
+      icons: Object.freeze({
+        positive: positiveIcon,
+        destroy: deleteIcon,
+        chevron: chevronIcon,
+        grip: gripIcon,
+      }),
+    };
+  },
+  props: {
+    disabled: {
+      type: Boolean,
+    },
+    items: {
+      type: Array,
+    },
+  },
+  methods: {
+    sortedChecklist () {
```

Comment:
is this necessary? It just calls another method

Reply:
no, at first this was the method from the task-modal, then I had to change the way to update it , so this'll be removed